elastic/kibana
# Add context.originalAlertState to the Inventory Threshold rule's recovery context

## Summary

_**This PR depends on #147928 being merged first**_

This PR adds the `ALERT_ACTION_GROUP` to the Alerts-As-Data documents for the Inventory Threshold Rule. It then uses that value from the alert document in the recovery context to set `context.originalAlertState`. This also adds `context.originalStateWasALERT` and `context.originalStateWasWARNING` to allow for conditional Mustache templates.

```
{{#context.originalAlertStateWasALERT}}
This is a recovery for an ALERT
{{/context.originalAlertStateWasALERT}}

{{#context.originalAlertStateWasWARNING}}
This is a recovery for an WARNING
{{/context.originalAlertStateWasWARNING}}
```

## How to test

1. Start Kibana and ingest some data (Metricbeat or whatever)
2. Create a Inventory Threshold Rule with a threshold that will trigger and then eventually recover
3. Add a server log action for the recovery action group with `{{context}}`
4. Save the rule
7. Observe the log message with `originalAlertState` as `ALERT`
